app/presenters/nav_presenter.rb

```diff
@@ -0,0 +1,57 @@
+class NavPresenter
+  def initialize(args)
+    @params = args[:params]
+    @current_user = args[:current_user]
+  end
+
+  def navs
+    return [] unless @current_user
+
+    [
+      {
+        label: I18n.t('nav.dashboard'),
+        path: %i[user dashboard],
+        enabled: true
+      },
+      {
+        label: I18n.t('nav.team_dashboard'),
+        path: %i[teams dashboard],
+        subnavs: [
+          { label: I18n.t('nav.tabs.team'),
+            path: %i[teams dashboard],
+            controllers: ['dashboard'],
+            actions: ['teams'] },
+          { label: I18n.t('nav.tabs.individuals'),
+            path: %i[individuals dashboard],
+            controllers: ['dashboard'],
+            actions: ['individuals'] }
+        ],
+        enabled: @current_user&.lead?
+      },
+      {
+        label: I18n.t('nav.admin_dashboard'),
+        path: %i[admin dashboard],
+        subnavs: [
+          { label: I18n.t('nav.tabs.admin_dashboard'),
+            path: %i[admin dashboard],
+            controllers: ['dashboard'],
+            actions: ['admin'] },
+          { label: I18n.t('nav.tabs.admin_users'),
+            path: %i[admin users],
+            controllers: ['admin/users'],
+            actions: %w[index new edit] },
+          { label: I18n.t('nav.tabs.admin_group_types'),
+            path: %i[admin group_types],
+            controllers: ['admin/group_types', 'admin/user_groups'],
+            actions: %w[index new edit] }
+        ],
+        enabled: @current_user&.admin?
+      }
+    ]
+  end
+
```

Comment:
Could add some more helpers here:

```suggestion
  def nav_active?(nav)
    path_active?(nav) || any_subnavs_active?(nav)
  end

  def path_active?(nav)
    nav[:path] == [@params['action'].to_sym, @params['controller'].to_sym]
  end

  def any_subnavs_active?(nav)
    nav[:subnavs]&.any? { |subnav| subnav_active?(subnav) }
  end
```

Reply:
I've got something very similar in the next PR so it's coming :-)